src/CommunityToolkit.Maui.Core/Layouts/DockLayoutManager.shared.cs

```diff
@@ -0,0 +1,156 @@
+﻿using CommunityToolkit.Maui.Core.Interfaces;
+using Microsoft.Maui.Layouts;
+
+namespace CommunityToolkit.Maui.Core.Layouts;
+
+/// <summary>LayoutManager for <see cref="IDockLayout"/>.</summary>
+public class DockLayoutManager : LayoutManager, IDockLayoutManager
+{
+	readonly IDockLayout dockLayout;
+
+	/// <summary>Ctor of DockLayoutManager.</summary>
+	public DockLayoutManager(IDockLayout dockLayout)
+		: base(dockLayout)
+	{
+		this.dockLayout = dockLayout;
+	}
+
+	/// <inheritdoc />
+	public override Size Measure(double widthConstraint, double heightConstraint)
+	{
+		var padding = dockLayout.Padding;
+		var spacing = dockLayout.Spacing;
+
+		double width = padding.HorizontalThickness;
+		double height = padding.VerticalThickness;
+
+		var widthLimit = widthConstraint - padding.HorizontalThickness;
+		var heightLimit = heightConstraint - padding.VerticalThickness;
+
+		foreach (var child in dockLayout)
+		{
+			if (child.Visibility != Visibility.Visible)
+			{
+				continue;
+			}
+
+			var childSize = child.Measure(widthLimit, heightLimit);
+
+			switch (dockLayout.GetDock(child))
+			{
+				case DockEnum.Left:
+				case DockEnum.Right:
+				{
+					var childWidth = childSize.Width + spacing.Width;
+					width += childWidth;
+					widthLimit -= childWidth;
+					break;
+				}
+				case DockEnum.Top:
+				case DockEnum.Bottom:
+				{
+					var childHeight = childSize.Height + spacing.Height;
+					height += childHeight;
+					heightLimit -= childHeight;
+					break;
+				}
+				case DockEnum.None:
+				default:
+				{
+					width += childSize.Width;
+					widthLimit -= childSize.Width;
+					height += childSize.Height;
+					heightLimit -= childSize.Height;
+					break;
+				}
+			}
+		}
+
+		return new Size(Math.Min(width, widthConstraint), Math.Min(height, heightConstraint));
+	}
+
+	/// <inheritdoc />
+	public override Size ArrangeChildren(Rect bounds)
+	{
+		var padding = dockLayout.Padding;
+		var spacing = dockLayout.Spacing;
+
+		var x = bounds.Left + padding.Left;
+		var y = bounds.Top + padding.Top;
+
+		var width = bounds.Width - padding.HorizontalThickness;
+		var height = bounds.Height - padding.VerticalThickness;
+
+		var i = 0;
+		foreach (var child in dockLayout)
```

Comment:
Why not use the `for` loop instead? Since we need the index

Reply:
Absolutely. Will do.